diff --git a/dist/parser.js b/dist/parser.js
index e0451de00f43e74f850785f297c85e5c0ec8acdd..2c851606ae18e182a32e17969d5166f077b0c9f4 100644
--- a/dist/parser.js
+++ b/dist/parser.js
@@ -122,10 +122,7 @@ function indexesOf(array, item) {
 }
 
 function uniqs() {
-  var list = Array.prototype.concat.apply([], arguments);
-  return list.filter(function (item, i) {
-    return i === list.indexOf(item);
-  });
+  return Array.from(new Set(Array.prototype.concat.apply([], arguments)));
 }
 
 var Parser = /*#__PURE__*/function () {
@@ -976,11 +973,14 @@ var Parser = /*#__PURE__*/function () {
     var interpolations = indexesOf(word, '#{');
 
     if (interpolations.length) {
+      var interpolationIndexes = new Set(interpolations);
       hasId = hasId.filter(function (hashIndex) {
-        return !~interpolations.indexOf(hashIndex);
+        return !interpolationIndexes.has(hashIndex);
       });
     }
 
+    var classIndexes = new Set(hasClass);
+    var idIndexes = new Set(hasId);
     var indices = (0, _sortAscending["default"])(uniqs([0].concat(hasClass, hasId)));
     indices.forEach(function (ind, i) {
       var index = indices[i + 1] || word.length;
@@ -995,14 +995,14 @@ var Parser = /*#__PURE__*/function () {
       var sourceIndex = current[_tokenize.FIELDS.START_POS] + indices[i];
       var source = getSource(current[1], current[2] + ind, current[3], current[2] + (index - 1));
 
-      if (~hasClass.indexOf(ind)) {
+      if (classIndexes.has(ind)) {
         var classNameOpts = {
           value: value.slice(1),
           source: source,
           sourceIndex: sourceIndex
         };
         node = new _className["default"](unescapeProp(classNameOpts, "value"));
-      } else if (~hasId.indexOf(ind)) {
+      } else if (idIndexes.has(ind)) {
         var idOpts = {
           value: value.slice(1),
           source: source,
